src/JustSaying/Messaging/Channels/IMultiplexer.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+using JustSaying.Messaging.MessageHandling;
+using JustSaying.Messaging.Monitoring;
+using Microsoft.Extensions.Logging;
+
+namespace JustSaying.Messaging.Channels
+{
+    internal interface IMultiplexer
+    {
+        Task Start();
+        void ReadFrom(ChannelReader<IQueueMessageContext> reader);
+        IAsyncEnumerable<IQueueMessageContext> Messages();
+    }
+
+    internal class RoundRobinQueueMultiplexer : IMultiplexer, IDisposable
```

Comment:
I don't think we need a multiplexer, having multiple consumers of a channel reader will result in fair distribution of reads, as channels has an internal queue for awaiting readers.

Reply:
Is that true for writes too? I'm less concerned about which reader takes a message from the queue and more about one writer/sqsqueue flooding the central queue.]
Something interesting to look into though 👍